uvacw/osd2f
# Feature/azure secret support

# Added support for Azure Keyvault variables

## Closes #58 
____
Some configuration settings are sensitive. OSD2F should support app deployments that use secrets, but not require any specific solution. This change adds support for implicit secret resolution in environment variables from the azure KeyVault service. 

Includes changes in #62 

## Assumptions
* assumes the `azure-keyvault::` prefix for environment variables is acceptable
* assumes the config file is always loaded

## Usage / Minimal Example

1. Test the integration locally
    * Make a keyvault
    * Create an arbitrary database filename as a `OSD2F-DB-URL` value in the keyvault
    * Start the app locally by passing the azure keyvault location of the secret as per the [secret_documention](https://github.com/uvacw/osd2f/blob/feature/azure-secret-support/docs/using_secret_stores.md) 
2. Test the integration on Azure
    * add a new secret DB-url that contains the value you would like to use in production
    * follow the [secret store documentation](https://github.com/uvacw/osd2f/blob/feature/azure-secret-support/docs/using_secret_stores.md) to give you existing webapp access to the keystore
    * change the `OSD2F-DB-URL` environment variable to the azure-keyvault reference of the new secret
    * deploy the code in this branch to the webapp following the [webapp deployment documentation](https://github.com/uvacw/osd2f/blob/feature/azure-secret-support/docs/deploying_to_azure.md)
    * observe the logs, everything works if the app starts without error. 

## Checklist
- [X] Added tests if appropriate (and it should always be)
- [X] Created new issues when required
